src/pretix/presale/templates/pretixpresale/event/base.html

```diff
@@ -111,9 +111,37 @@ <h1>
     {% if request.event.testmode %}
         {% if request.sales_channel.testmode_supported %}
             <div class="alert alert-warning">
-                <p><strong><span class="sr-only">{% trans "Warning" context "alert-messages" %}:</span>
-                    {% trans "This ticket shop is currently in test mode. Please do not perform any real purchases as your order might be deleted without notice." %}
+                <p><strong>
+                    <span class="sr-only">{% trans "Warning" context "alert-messages" %}:</span>
+                    {% trans "This ticket shop is currently in test mode." %}
                 </strong></p>
+                <p>
+                    {% trans "Please do not perform any real purchases as your order might be deleted without notice." %}
+                </p>
+                {% if request.now_dt_is_fake %}
+                    <p>
+                        {% blocktrans trimmed with datetime=request.now_dt|date:"SHORT_DATETIME_FORMAT" %}
+                        You are currently using the time machine. The ticket shop is rendered as if it were {{ datetime }}.
+                        {% endblocktrans %}
+                        <a href="{% eventurl event "presale:event.timemachine" %}" class="stealth"><span class="fa fa-clock-o" aria-hidden="true"></span> <u>{% trans "Change" %}</u></a>
+                    </p>
+                {% elif request.user.is_authenticated or request.event_access_user.is_authenticated %}
+                    <p>
+                        {% eventurl event "presale:event.timemachine" as time_machine_link %}
+                        {% blocktrans trimmed with time_machine_link=time_machine_link %}
+                            To view your shop at different points in time, you can enable
+                            <a href="{{ time_machine_link }}" class="stealth"><span class="fa fa-clock-o" aria-hidden="true"></span> <u>time machine</u></a>.
+                        {% endblocktrans %}
+                    </p>
+                {% elif request.event_domain or request.organizer_domain %}
+                    <p>
+                        {% url "control:event.transfer_session" event=event.slug organizer=event.organizer.slug as time_machine_link %}
```

Comment:
This does not seem to work, ``{% url %}`` usually generates relative URLs, so this will link to /control, but we would need to link to the main domain (from settings.SITE_URL)

Reply:
It works for me, in a somewhat ugly way though: It links to `"/control/event/.../transfer_session/"`, so the browser requests `https://customdomain/control/event/.../transfer_session/`, which gets in turn redirected to `https://maindomain/control/event/.../transfer_session/`. Is this different on your machine?

This seems to be the same behaviour as the "View in backend" button on the order detail page.

I'm not quite sure what the best way to fix this would be: Currently, settings (and therefore SITE_URL) is not available in the default context of presale. 
- I could change that and just use `time_machine_link=settings.SITE_URL|add:time_machine_link`. 
- Or I could try and fix the `url` template tag (or, more realistically, create a new derived one) which always calls build_absolute_url (or do we have something like that already?).

Recommendations?

Reply:
Added a new absmainurl template tag based on the other multidomain url template tags